### ISSUE TITLE
[onert] In-place Reshape and ExpandDims operation 

### ISSUE THREAD

**Issue by hseok-oh** [Samsung/ONE#12836]
We are always using `memcpy` for `Reshape` and `ExpandDims` operation.
If these operation's input is used only once and updated shape does not require padding, we can skip copy operation. 

For this, we need to modify memory allocation and tensor handling implementation.

**Comment by mbencer** [comment]
Due to recommendation from https://github.com/Samsung/ONE/issues/12814#issuecomment-2316619907 I'm assigning myself to this issue ;)

**Comment by mbencer** [comment]
@hseok-oh Could you please take a look at https://github.com/Samsung/ONE/pull/14057?
I am not sure if you plan implementation similar to my draft or maybe more global approach (like creating common buffer during tensor creation). The second one approach has much larger affected area.

**Comment by mbencer** [comment]
@hseok-oh Handling of sharing memory for static shapes is already on master. Should dynamic shape support be processed also under this issue or is it better to create a separate issue for it?
BTW.: Should I treat dynamic shapes part as something with big priority?

**Comment by hseok-oh** [comment]
>@hseok-oh Handling of sharing memory for static shapes is already on master. Should dynamic shape support be processed also under this issue or is it better to create a separate issue for it?

It doesn't matter either way.

BTW.: Should I treat dynamic shapes part as something with big priority?

Treating dynamic shape is more useful. But considering its impact on performance, It may not has a very high priority.
